src/counts.jl

```diff
@@ -388,6 +390,8 @@ of occurrences.
 """
 countmap(x::AbstractArray{T}; alg = :auto) where {T} = addcounts!(Dict{T,Int}(), x; alg = alg)
 countmap(x::AbstractArray{T}, wv::AbstractVector{W}) where {T,W<:Real} = addcounts!(Dict{T,W}(), x, wv)
+countmap(x) = addcounts!(Dict{eltype(x),Int}(), x; alg = :auto)
+countmap(x, wv::AbstractVector{W}) where {W<:Real} = addcounts!(Dict{eltype(x),W}(), collect(x), wv)
```

Comment:
As I said currently the policy followed across the package is that we only provide weighted methods when both inputs are vectors. Better stick to that.
```suggestion
```